### DISCUSSION TITLE
LocalSend Speed is just 1-2mbps while transferring files from windows to android.

### DISCUSSION THREAD

**Discussion by udayramp** [localsend/localsend#2045] [Discussion (English, International)]
I have a 5GHz hotspot active on my mobile and no encryption active on both devices. While transferring from mobile to PC, it gives me a speed of 30-40mbps.

**Comment by xihale** [comment]
Turn secure(https, tls) option off.

**Reply by Deathwing1999** [reply]
The same goes for turning it off.
If it comes out at a normal speed when transferring from Android to Windows in the first place, but it only slows down when going from Windows to Android, wouldn't it be a Secure setup problem?

**Reply by xihale** [reply]
To be honest, I don't really know, but I think it would be a programming bug according to what you had shown.

**Reply by Tienisto** [reply]
Do you use a custom destination folder?

**Reply by Deathwing1999** [reply]
By default, you use a custom folder, and there is no significant change when you switch to a download folder.

**Comment by Deathwing1999** [comment]
I'm experiencing the same problem.
Modifying security settings does not produce any significant results.
When you use FTP tools such as WinSCP, you get a good speed when transferring from your PC to Android.
Roughly 30 to 40 MB/s

**Reply by black-backdoor** [reply]
Hi,
I think this has to do with the way LocalSend works. As far as I understand, it uses HTTP POST requests to upload each file in a single go. I don't know exactly how the protocol is implemented in this repository (I haven’t looked at it), but depending on the server implementation, this could require loading the entire file into memory. In extreme cases, this could consume a significant amount of RAM, which might even trigger the operating system to take measures to stop memory overuse, such as killing the process or throttling resources.

This is why I’m proposing [localsend/protocol#31](https://github.com/localsend/protocol/issues/31) - to switch to a download-based approach instead.
